packages/localize/src/number/forceTryCurrencyCode.js

```diff
@@ -4,13 +4,16 @@
  * @param {import('../../types/LocalizeMixinTypes').FormatNumberOptions} [options]
  * @returns {FormatNumberPart[]}
  */
+
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-unused-expressions */
 export function forceTryCurrencyCode(formattedParts, { currency, currencyDisplay } = {}) {
   const result = formattedParts;
-  // Change the currency code from TRY to TL, for Turkey
+  // Change the currency code from TRY to TL
   if (currency === 'TRY' && currencyDisplay === 'code') {
-    if (result[0].value === 'TRY') {
-      result[0].value = 'TL';
-    }
+    result.forEach(part => {
+      part.type === 'currency' ? (part.value = 'TL') : '';
+    });
```

Comment:
this code seems a little hard to read (to me) 😅

could you add a test for it?

and afterward, potentially change it to something like this?

```suggestion
    return formattedParts.map(part => {
      const newPart = part;
      if (part.type === 'currency') {
         newPart.value = 'TL';
      }
      return newPart;
    });
```

this would also mean that both eslint disabled would not be needed 💪

Reply:
Thank you :) But with map, objects have immutability therefor I used forEach function to modify array parts.. And also I didn't want to use object assign and create a new array of parts because it would slow things down.  Maybe if I changed arrow function to default function and used "if" statement, would it help?

Reply:
Thanks for your review. I made it easier to read and added tests for both positive and negative numbers.